hcl/hcl_script.go

```diff
@@ -0,0 +1,211 @@
+			if err != nil {
+				errs.Append(err)
+			}
+			parsedScript.Jobs = append(parsedScript.Jobs, parsedJobBlock)
+		default:
+			errs.Append(errors.E(ErrScriptUnknownBlock, nestedBlock.TypeRange, nestedBlock.Type))
+
+		}
+	}
+
+	if err := errs.AsError(); err != nil {
+		return nil, err
+	}
+
+	return parsedScript, nil
+
+}
+
+func (p *TerramateParser) validateDescription(attr ast.Attribute) (ast.Attribute, error) {
+	errs := errors.L()
+	val, diags := attr.Expr.Value(nil)
+	if diags.HasErrors() {
+		errs.Append(diags)
+		return ast.Attribute{}, errs.AsError()
+	}
+
+	if val.AsString() == "" {
+		errs.Append(errors.E("empty description"))
+		return ast.Attribute{}, errs.AsError()
+	}
+
+	return attr, nil
+}
+
+func validateScriptJobBlock(block *ast.Block) (*ScriptJob, error) {
+	errs := errors.L()
+
+	var foundCmd, foundCmds bool
+	parsedScriptJob := &ScriptJob{}
+	for _, attr := range block.Attributes {
+		switch attr.Name {
+		case "command":
+			cmdAttr, err := validateCommand(attr)
+			if err != nil {
+				errs.Append(errors.E(ErrScriptInvalidCmd, attr.NameRange, attr.Name))
+				continue
+			}
+			parsedScriptJob.Command = cmdAttr
+			foundCmd = true
+		case "commands":
+			cmdsAttr, err := validateCommands(attr)
+			if err != nil {
+				errs.Append(errors.E(ErrScriptInvalidCmds, attr.NameRange, attr.Name))
+				continue
+			}
+			parsedScriptJob.Commands = cmdsAttr
+			foundCmds = true
+		default:
+			errs.Append(errors.E(ErrScriptUnknownAttr, attr.NameRange, attr.Name))
+
+		}
+	}
+
+	// job.command and job.commands are mutually exclusive
+	if foundCmd && foundCmds {
+		errs.Append(errors.E(ErrScriptCmdConflict, block.TypeRange))
+	}
+
+	if err := errs.AsError(); err != nil {
+		return nil, err
+	}
+
+	return parsedScriptJob, nil
+}
+
+// validateCommand validates the provided script job block, parses the attribute
+// into Command and returns an error if validation fails
+func validateCommand(cmdAttr ast.Attribute) (*Command, error) {
+	errs := errors.L()
+	val, diags := cmdAttr.Attribute.Expr.Value(nil)
+	if diags.HasErrors() {
+		errs.Append(diags)
+	}
+
+	_, err := ValueAsStringList(val)
+	if err != nil {
+		errs.Append(err)
+	}
+
+	if err := errs.AsError(); err != nil {
+		return nil, err
+	}
+
+	parsed := Command(cmdAttr)
+
+	return &parsed, nil
+
+}
+
+// validateCommands validates the provided cmdsAttr, parses the attribute into
+// Commands and returns an error if validation fails
+func validateCommands(cmdsAttr ast.Attribute) (*Commands, error) {
+	errs := errors.L()
+	val, diags := cmdsAttr.Attribute.Expr.Value(nil)
+	if diags.HasErrors() {
+		errs.Append(diags)
+	}
+
+	if !val.Type().IsTupleType() {
+		errs.Append(errors.E("wrong type"))
+	}
+
+	valSlice := val.AsValueSlice()
+	for _, val := range valSlice {
+		_, err := ValueAsStringList(val)
+		if err != nil {
+			errs.Append(err)
+		}
+	}
+
+	if err := errs.AsError(); err != nil {
+		return nil, err
+	}
+
+	parsed := Commands(cmdsAttr)
+	return &parsed, nil
```

Comment:
same as for `validateDescription()`. We cannot evaluate them at this time.

Reply:
Fixed in https://github.com/terramate-io/terramate/pull/1236/commits/8e6f17a968d313064e25a3be6d9d0e715f6e34be